### PR TITLE
Feat / live channels screen with EPG view

### PR DESCRIPTION
**Pull request by ChristiaanScheermeijer**
**This PR will be used as the "main" branch for the EPG feature**

@marcovandeveen I've mapped the scope from #24 to the planned features. Let me know if you need changes here.

### Must have

- ✅ Desktop + mobile support
- ✅ EPG data delivery from a remote URL
- ✅ Multiple channels (3-5 channels is sweet spot)
- ✅ Visualize EPG in local timezone
- ✅ Broadcast Live 24x7 Streams
- ✅ Play current live items
- ✅ Scroll back in the current program (DVR - default 5 mins) (managed from the streaming dashboard)

### Should have

- ✅ Play historical live items (Catchup - default 8 hours)
- ✅ Deal with EPG longer than catchup service
- ⚠️ Scroll through future / past (max 7 days) (multiple days don't work well in EPG)
- ✅ See in OTT Analytics that the 'playlist' source was EPG
- ✅ Demo mode
- ✅ Support for live streams without valid EPG data / URL (~show continuous line~ empty timeline)
- 🅿️ Link to JW VOD item OR series item (parked)
- ✅ Start from the beginning

### Nice-to-have

- 🅿️ Right-to-live support text of text
- 🅿️ Select (long-term) future date
- ✅ Program-level images
- 🅿️ Translation support
- 🅿️ Option to visualize EPG in an absolute timezone

📝 = planned
🅿️ = not planned
🏗️ = in progress
⚠️ = impediment(s)
✅ = finished

### CODE REVIEW

**Comment by marcovandeveen** [comment]
Sorry, pressed "ready for review" button on accident!

**Comment by marcovandeveen** [comment]
Wow Christiaan, you are aiming high with this planned scope!

Note about "See in OTT Analytics that the 'playlist' source was EPG"
In the current solution direction, it means feeding [analytics parameter](https://github.com/jwplayer/ott-web-app/blob/6304c8031efe92738691b2cabe95c07bb7500b13/docs/features/video-analytics.md#session-metrics) `fed` (Feed ID) attribute with the EPG source playlist.

**Comment by marcovandeveen** [comment]
@ChristiaanScheermeijer 

@AntonLantukh highlighted an item we didn't discuss: "open live channel from homepage shelf". It's currently opening the movie page. But should now open the "live" page, with the right channel selected.  

Could be based on scheduleURL (similar to opening series, based on seriesId).

I added to the 'must have' scope. Could you factor this into your plan? I think it's more important than having images in the schedule or Link to JW VOD item OR series item. (in case we need to choose)

**Comment by ChristiaanScheermeijer** [comment]
Thanks @marcovandeveen 

Ah, then I guess adding the OTT analytics `fed` property is an easy win. I will add it to the playback tickets 💪 

When adding live media items to the homepage, we'll be missing the containing playlist id. This is needed in order to get all channels. There is no concrete reference to this value at the moment.

However, if you add the Live playlist as a shelf, we can deep link to the live screen with the live channel selected.

But it would also be nice when it's possible to promote live channels in the featured shelf...

**Comment by ChristiaanScheermeijer** [comment]
Note, that tests are failing because I've added two channels to the featured playlist.

**Comment by ChristiaanScheermeijer** [comment]
@dbudzins we'll be testing this deploy preview tomorrow. In addition, @RCVZ will submit a PR this evening or tomorrow as well with some automated tests for the live channels page.

Most of the code is already reviewed in the previous PRs, but please let us know if you have feedback.

File: src/i18n/locales/en_US/epg.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "on_channel": "On {{name}}",
+  "start_from_beginning": "Watch from begin"
```
Comment (budzworthy) [inline src/i18n/locales/en_US/epg.json]: This doesn't sound quite right (not sure why it's not reflected on the preview url)

File: src/containers/PlaylistLiveChannels/PlaylistLiveChannels.tsx [hunk 2]
```diff
@@ -128,19 +129,40 @@ function PlaylistLiveChannels({ playlist: { feedid, title, playlist } }: { playl
...
-  const pageTitle = `${title} - ${siteName}`;
+  // SEO (for channels)
+  const canonicalUrl = `${window.location.origin}${liveChannelsURL(feedid, channel.id)}`;
+  const pageTitle = `${channel.title} - ${siteName}`;
 
   return (
     <>
       <Helmet>
         <title>{pageTitle}</title>
+        <link rel="canonical" href={canonicalUrl} />
```
Comment (budzworthy) [inline src/containers/PlaylistLiveChannels/PlaylistLiveChannels.tsx:145]: This big blob of meta tags is repeated almost identically in 3 places. Would it be better placed in the VideoDetails component?

File: src/i18n/locales/en_US/epg.json (hunk 1 above)
Comment (ChristiaanScheermeijer) [reply src/i18n/locales/en_US/epg.json]: Yes, you're right! According to @marcovandeveen it should be `Watch from start`

File: src/containers/PlaylistLiveChannels/PlaylistLiveChannels.tsx (hunk 2 above)
Comment (ChristiaanScheermeijer) [reply src/containers/PlaylistLiveChannels/PlaylistLiveChannels.tsx:145]: Yes, that's a valid point. I think we can refactor this into a single SEO/Metatags component. I will create a ticket for it.

Comment (ChristiaanScheermeijer) [reply src/containers/PlaylistLiveChannels/PlaylistLiveChannels.tsx:145]: Created ticket #139

**Review by budzworthy — APPROVED** [review]

**Pull request merged**